lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        handler = this.handler;
+        handler = this.handler,
+        serializer = this.serializer || (handler && handler.serialize);
```

Comment:
I'm in agreement with @MiguelMadero here. Any route which gets lazy-loaded should have its `serialize` hook replaced with an assertion, or fail to build. I'm pretty sure that we shouldn't handle that as part of router.js, though.